### PR TITLE
Changed files issues

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java [hunk 1]
```diff
@@ -202,7 +202,7 @@ private static boolean exists(@Nullable VirtualFile file) {
       return false;
     }
 
-    file.refresh(false, false);
+    // file.refresh(false, false);
```
Comment (henryju) [inline src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java]: Is it expected to stay?

File: src/main/java/org/sonarlint/intellij/messages/ChangedFilesIssuesListener.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package org.sonarlint.intellij.messages;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.messages.Topic;
+import java.util.Collection;
+import java.util.Map;
+import org.sonarlint.intellij.issue.LiveIssue;
+
+public interface ChangedFilesIssuesListener {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/messages/ChangedFilesIssuesListener.java]: CRITICAL Move constants to a class or enum. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1214)
CRITICAL Annotate the "ChangedFilesIssuesListener" interface with the @FunctionalInterface annotation [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

File: src/main/java/org/sonarlint/intellij/messages/GlobalConfigurationListener.java [hunk 3]
```diff
@@ -21,11 +21,32 @@
 
 import com.intellij.util.messages.Topic;
 import java.util.List;
+import java.util.function.BiConsumer;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/messages/GlobalConfigurationListener.java]: MINOR Remove this unused import 'java.util.function.BiConsumer'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/messages/GlobalConfigurationListener.java [hunk 4]
```diff
@@ -21,11 +21,32 @@
...
+import java.util.function.Consumer;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/messages/GlobalConfigurationListener.java]: MINOR Remove this unused import 'java.util.function.Consumer'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java [hunk 5]
```diff
@@ -48,36 +52,61 @@ public OpenFilesSubmitter(Project project, SonarLintConsole console, FileEditorM
...
+    CompletableFuture<AnalysisResult> future = new CompletableFuture<>();
+    future.complete(new AnalysisResult(0, Collections.emptyMap()));
+    return future;
   }
 
-  private void submitFiles(VirtualFile[] files, TriggerType trigger) {
+  private Multimap<Module, VirtualFile> filterAndgetByModule(VirtualFile[] files, boolean autoTrigger) {
     Multimap<Module, VirtualFile> filesByModule = HashMultimap.create();
 
     for (VirtualFile file : files) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java]: MAJOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS135)

File: src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java [hunk 6]
```diff
@@ -26,19 +26,23 @@
 import com.intellij.openapi.module.Module;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/trigger/SonarLintSubmitter.java]: MINOR Remove this unused import 'java.util.concurrent.Future'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/ui/LastAnalysisPanel.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    panel.add(Box.createHorizontalBox(), gc);
+  }
+
+  private void createComponents() {
+    panel = new JPanel(new GridBagLayout());
+    try {
+      icon = new JLabel(ResourceLoader.getIcon("info.png"));
+    } catch (IOException e) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/LastAnalysisPanel.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java [hunk 8]
```diff
@@ -32,6 +34,7 @@
 import org.sonarlint.intellij.core.ProjectBindingManager;
 import org.sonarlint.intellij.core.SonarLintFacade;
 import org.sonarlint.intellij.ui.SonarLintConsole;
+import org.sonarlint.intellij.util.SonarLintUtils;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java]: MINOR Remove this unused import 'org.sonarlint.intellij.util.SonarLintUtils'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java [hunk 9]
```diff
@@ -21,6 +21,8 @@
 
 import com.intellij.openapi.application.AccessToken;
 import com.intellij.openapi.application.Application;
+import com.intellij.openapi.application.ApplicationManager;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/analysis/SonarLintAnalyzer.java]: MINOR Remove this unused import 'com.intellij.openapi.application.ApplicationManager'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+public class SonarLintChangedPanel extends SimpleToolWindowPanel implements OccurenceNavigator, DataProvider {
+  private static final String ID = "SonarLint";
+  private static final String GROUP_ID = "SonarLint.changedtoolwindow";
+  private static final String SPLIT_PROPORTION = "SONARLINT_CHANGED_ISSUES_SPLIT_PROPORTION";
+
+  private final Project project;
+  private final SonarLintRulePanel rulePanel;
+  private final LastAnalysisPanel lastAnalysisPanel;
+  private Tree tree;
+  private ActionToolbar mainToolbar;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java]: CRITICAL Make "mainToolbar" transient or serializable. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+  private TreeModelBuilder treeBuilder;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java]: CRITICAL Make "treeBuilder" transient or serializable. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+import org.sonarlint.intellij.issue.ChangedFilesIssues;
+import org.sonarlint.intellij.messages.ChangedFilesIssuesListener;
+import org.sonarlint.intellij.messages.StatusListener;
+import org.sonarlint.intellij.ui.nodes.AbstractNode;
+import org.sonarlint.intellij.ui.nodes.IssueNode;
+import org.sonarlint.intellij.ui.tree.IssueTree;
+import org.sonarlint.intellij.ui.tree.TreeModelBuilder;
+import org.sonarlint.intellij.util.SonarLintUtils;
+
+public class SonarLintChangedPanel extends SimpleToolWindowPanel implements OccurenceNavigator, DataProvider {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/ui/SonarLintChangedPanel.java]: MAJOR This class has 6 parents which is greater than 5 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
MINOR "DataProvider" is implemented by a super class; there is no need to implement it here. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1939)

File: src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeChangedFilesAction.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+package org.sonarlint.intellij.actions;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.changes.ChangeListManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/actions/SonarAnalyzeChangedFilesAction.java]: MINOR Remove this unused import 'java.util.concurrent.Future'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 22 issues
* BLOCKER 1 blocker
* CRITICAL 7 critical
* MAJOR 5 major
* MINOR 9 minor

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 12 issues
* CRITICAL 6 critical
* MAJOR 4 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues
* CRITICAL 6 critical
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**